### DISCUSSION TITLE
Pyrra support

### DISCUSSION THREAD

**Discussion by mskarbek** [grafana/mimir#5283] [General]
I would like to use [Pyrra](https://github.com/pyrra-dev/pyrra) in my env but after a quick test, Pyrra seems not to recognize the Prometheus endpoint in Mimir as a correct one/compatible. Since Mimir already tries to provide Prometheus endpoint (with some caveats and quirks) for Grafana, I was wondering if pursuing compatibility on a level sufficient for Pyrrra would be something of an interest for Mimir maintainers?
Currently, I'm migrating to Grafana/Mimir/Loki/Vector stack and reintroduction of Prometheus would be undesirable. Since Mimir doesn't provide a remote read endpoint, I would have to duplicate sinks in Vector to provide data to Prometheus instead of pointing Prometheus at Mimir as a remote read source and use it as a proxy for Pyrra.

**Comment by pstibrany** [comment]
Hi, thank you for asking. In general, we are happy to accept contributions improving compatibility with Prometheus API, but it also depends on which changes are needed.

Do you know what's missing in Mimir to make Pyrra work?

> Since Mimir doesn't provide a remote read endpoint

Mimir exposes remote read endpoint already: https://grafana.com/docs/mimir/latest/references/http-api/#remote-read

**Reply by mskarbek** [reply]
Ok, I'll play with Prometheus as proxy with remote read as a starting point, and then try to figure out what Pyrra actually requires to work properly with only Mimir as a data provider.

**Reply by ItayReshef** [reply]
Hey!, I would love to hear if you discovered anything or reached a solution. I'm after the exact same thing.

**Comment by 56quarters** [comment]
If you can figure out what Pyrra is looking for, please open an issue and we can possibly implement something. We've done something similar for Kiali in the past: https://github.com/grafana/mimir/pull/3983 https://github.com/grafana/mimir/issues/3323

**Comment by jakeforge** [comment]
I was able to get this working by appending '/prometheus' to the mimir gateway url i.e.
`http://mimir-gateway.namespace.svc.cluster.local/prometheus` @ItayReshef @mskarbek

**Reply by juliendf** [reply]
Hi, I've tried this solution but it does not work for me I have HTTP 429 errors. 
Are you running a multi tenant infrastructure requiring X-Scope-OrgID header ? I wonder if this is my problem
